### PR TITLE
use per contact queues for handling

### PR DESCRIPTION
**Pull request by nicpottier**

### CODE REVIEW

File: temba/msgs/tasks.py [hunk 1]
```diff
@@ -48,23 +49,11 @@ def process_run_timeout(run_id, timeout_on):
...
-    # somebody already handled this message, move on
-    if not msg:  # pragma: needs cover
-        return
-
-    # get a lock on this contact, we process messages one by one to prevent odd behavior in flow processing
-    key = 'pcm_%d' % msg.contact_id
-
-    # wait for the lock as we want to make sure to process the next message as soon as we are free
-    with r.lock(key, timeout=120):
+    if msg:
```
Comment (rowanseymour) [inline temba/msgs/tasks.py]: Can `msg` be falsey ?

File: temba/msgs/tasks.py [hunk 2]
```diff
@@ -78,6 +67,44 @@ def process_message_task(msg_id, from_mage=False, new_contact=False):
...
+        with r.lock(key, timeout=120):
+            # pop the next message to process off our contact queue
+            with r.pipeline() as pipe:
+                pipe.zrange(contact_queue, 0, 0)
+                pipe.zremrangebyrank(contact_queue, 0, 0)
+                (contact_msg, deleted) = pipe.execute()
+
+            if contact_msg:
+                msg_event = json.loads(contact_msg[0])
+                msg = Msg.objects.filter(pk=msg_event['id'], status=PENDING).select_related('org', 'contact', 'contact_urn', 'channel').first()
```
Comment (rowanseymour) [inline temba/msgs/tasks.py]: `id=msg_event['id']`

File: temba/msgs/models.py [hunk 3]
```diff
@@ -1109,6 +1112,19 @@ def update(self, cmd):
 
         return handled
 
+    def queue_handling(self):
+        """
+        Queues this message to be handled by one of our celery queues
+        """
+        payload = dict(type=MSG_EVENT, contact_id=self.contact.id, id=self.id, from_mage=False, new_contact=False)
```
Comment (rowanseymour) [inline temba/msgs/models.py]: If we want a consistent schema for events placed in a per contact queue then just thinking maybe `id` should be the contact id, and the message id be `msg_id`

File: temba/msgs/tasks.py (hunk 1 above)
Comment (nicpottier) [inline temba/msgs/tasks.py]: Doesn't look that way, will remove.

File: temba/msgs/models.py (hunk 3 above)
Comment (nicpottier) [inline temba/msgs/models.py]: In time agree, though right now for backwards compatibility need to keep that guy. (in time the stuff going in the actual handler queue can be JUST the contact_id)

Comment (rowanseymour) [inline temba/msgs/models.py]: Ah ok thought this was a new thing

**Pull request merged**